main.tf

```diff
@@ -73,6 +73,17 @@ resource "aws_security_group" "redis" {
   }
 }
 
+resource "aws_security_group_rule" "redis_ingress_self" {
+  count = var.ingress_self == true ? 1 : 0
```

Comment:
`var.ingress_self` is a boolean so you can write it as follows:
```
count = var.ingress_self ? 1 : 0
```

Reply:
Fixed. Thanks.